solr/core/src/java/org/apache/solr/handler/admin/SystemInfoHandler.java

```diff
@@ -139,8 +140,19 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
     if (solrCloudMode) {
       rsp.add("zkHost", getCoreContainer(req, core).getZkController().getZkServerAddress());
     }
-    if (cc != null)
-      rsp.add( "solr_home", cc.getSolrHome());
+    if (cc != null) {
+      String solrHome = cc.getSolrHome();
+      rsp.add("solr_home", solrHome);
+
+      Path coreRootDirectory = cc.getCoreRootDirectory();
+      if (coreRootDirectory != null) {
+        String coreRootDirectoryString = coreRootDirectory.toString();
+        if (!coreRootDirectoryString.equals(solrHome)) {
```

Comment:
I disagree with this condition.  I think this should always emitted no matter where it is.  It's more helpful on consumers to depend on its existence rather than having it vary in some cases but not others.

Reply:
Ok, I was on borderline myself. I felt that because this option is barely used, having the same long string sent twice was bit of a waste of bandwidth. But, I guess it does not happen too often.

I can correct both generation and invocation of this option to make it a guaranteed setup. Let's just hope they don't run version 9 tool against version 8 core.

Also, if you don't like the name (the other open point), this is a good time to mention it.

Reply:
The "solr_" part should be skipped IMO.  Everything here is Solr :-)

> Let's just hope they don't run version 9 tool against version 8 core.

Well it's up to them to have a tolerant client.  For the SolrCLI I think it makes sense to check for null in case of a cross-version situation.